src/data/set/basic.lean

```diff
@@ -691,7 +689,7 @@ theorem insert_union : insert a s ∪ t = insert a (s ∪ t) := ext $ λ x, or.a
 
 @[simp] theorem union_insert : s ∪ insert a t = insert a (s ∪ t) := ext $ λ x, or.left_comm
 
-theorem insert_nonempty (a : α) (s : set α) : (insert a s).nonempty := ⟨a, mem_insert a s⟩
+@[simp] theorem insert_nonempty (a : α) (s : set α) : (insert a s).nonempty := ⟨a, mem_insert a s⟩
```

Comment:
One of those conflicts in naming conventions is that some think that since this could be written as `set.nonempty (insert a s)` that the correct name is `set.nonempty_insert`, but others think that since this is a predicate that would tend to be used with dot notation that it should be `set.insert_nonempty`.

Maybe this should be a Zulip poll. (Or maybe we have both with one an alias of the other -- some programming languages like Ruby have aliases to make things easier to find.)

Reply:
`insert_nonempty` matches other lemmas like `univ_nonempty` and `singleton_nonempty`. Moreover, it's the most common one of the two throughout `mathlib`. So this decision looks relatively clear-cut.

Reply:
Also, if I wanted to look this theorem up, I'd start from the more specific `insert`, followed by the more general `nonempty`.

Reply:
That might be what you'd do, but my point is that this is a long-standing conflict in the mathlib naming conventions, and it hasn't been resolved yet as far as I can tell. There's a tendency to go for prop-valued predicates being at the end of a name, but it'd be nice if it finally made it into [#naming](https://leanprover-community.github.io/contribute/naming.html).

This is one of a few posts (I'm finding these to be difficult to dig up; there's another I remember with a poll by Floris): https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/.23naming.20and.20dot.20notation/near/246520191

Reply:
I'll make the Zulip poll then.

Reply:
https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/naming.20contest/near/287004373

Reply:
Seems like the current name won out.